src/migrations/20230807000000-genericReport-migrateTrainingReports.js

```diff
@@ -0,0 +1,913 @@
+        )
+        SELECT
+          reports_id,
+          (data->>'numberOfParticipants')::int,
+          "createdAt",
+          "updatedAt"
+        FROM interim_reports
+        WHERE NULLIF(data->>'numberOfParticipants','') IS NOT NULL
+        ;
+
+        -- Populate ReportNextSteps
+        INSERT INTO "ReportNextSteps" (
+          "reportId",
+          note,
+          "noteType",
+          "completedDate",
+          "createdAt",
+          "updatedAt"
+        )
+        SELECT
+          reports_id,
+          data->'recipientNextSteps'->0->>'note',
+          '${NEXTSTEP_NOTETYPE.RECIPIENT}'::"enum_ReportNextSteps_noteType",
+          TO_DATE(NULLIF((data->'recipientNextSteps'->0->>'completeDate'),''),'MM/DD/YYYY'),
+          "createdAt",
+          "updatedAt"
+        FROM interim_reports
+        WHERE COALESCE((data->'recipientNextSteps'->0->>'note'),'') != ''
+        UNION
+        SELECT
+          reports_id,
+          data->'specialistNextSteps'->0->>'note',
+          '${NEXTSTEP_NOTETYPE.SPECIALIST}'::"enum_ReportNextSteps_noteType",
+          TO_DATE(NULLIF((data->'specialistNextSteps'->0->>'completeDate'),''),'MM/DD/YYYY'),
+          "createdAt",
+          "updatedAt"
+        FROM interim_reports
+        WHERE COALESCE((data->'specialistNextSteps'->0->>'note'),'') != ''
+        ;
+
+        -- Populate ReportPageStates
+        INSERT INTO "ReportPageStates" (
+          "reportId",
+          "pageState",
+          "createdAt",
+          "updatedAt"
+        )
+        SELECT
+          reports_id,
+          data->'pageState',
+          "createdAt",
+          "updatedAt"
+        FROM interim_reports
+        WHERE data->'pageState' IS NOT NULL
+        ;
+
+        -- Populate ReportImports
+        INSERT INTO "ReportImports" (
+          "reportId",
+          data,
+          "createdAt",
+          "updatedAt"
+        )
+        SELECT
+          reports_id,
+          erp.imported,
+          erp."createdAt",
+          erp."updatedAt"
+        FROM "EventReportPilots" erp
+        JOIN interim_reports ir
+          ON pilot_record_id = erp.id
+        WHERE imported IS NOT NULL
+        ;
+
+        ------------------------------------------------------------------------------------------------
+        ----- REPORT LINK TABLES -----------------------------------------------------------------------
+        ------------------------------------------------------------------------------------------------
+
+        -- Skip ReportApprovals: no approvals for TRs
+
+        -- Reused Collaborator tables
+        DROP TABLE IF EXISTS owner_status;
+        CREATE TEMP TABLE owner_status
+        AS
+        SELECT s.id sid
+        FROM "Statuses" s
+        JOIN "ValidFor" v
+          ON s."validForId" = v.id
+          AND v.name = 'collaborator'
+        WHERE s.name = 'approved' -- TODO: this is terminal, which could lock editing, but the others statuses make even less sense for events
```

Comment:
@hardwarehuman The collaborator statusId should be left null for any non-approving user